### PR TITLE
Small fixes to LICENSE and NOTICE files

### PR DESCRIPTION
**Pull request by massdosage**
While working on https://github.com/apache/iceberg/pull/1267 I noticed a few small things in the `NOTICE` and `LICENSE` files that I thought were worth fixing.

### CODE REVIEW

File: spark-runtime/LICENSE [hunk 1]
```diff
@@ -283,7 +283,7 @@ License: Apache License Version 2.0 http://www.apache.org/licenses/LICENSE-2.0
 This binary artifact contains fastutil.
 
 Copyright: 2002-2014 Sebastiano Vigna
-Home page: http://fasutil.di.unimi.it/
+Home page: http://fastutil.di.unimi.it/
```
Comment (massdosage) [inline spark-runtime/LICENSE:286]: There was a typo in the URL

File: spark-runtime/NOTICE [hunk 2]
```diff
@@ -153,6 +153,8 @@ notice:
 | standalone and requires a support library to be linked with it.  This
 | support library is itself covered by the above license.
 
+--------------------------------------------------------------------------------
```
Comment (massdosage) [inline spark-runtime/NOTICE:156]: There was a missing separator between these sections.

**Comment by massdosage** [comment]
@rdblue hopefully a quick, easy one to review!

**Comment by rdblue** [comment]
Looks great, thanks for fixing these!

**Pull request merged**